src/plugins/intellisense/index.tsx

```diff
@@ -0,0 +1,512 @@
+        ) {
+          this.partialFunctionCallDoc = current.text;
+          found = true;
+        }
+      }
+      if (!found) this.partialFunctionCallDoc = undefined;
+
+      // determine where to put intellisense window
+      const bbox = getMQCursorPosition(focusedMQ);
+      this.x = bbox?.left ?? 0;
+      this.y = bbox?.top ?? 0;
+
+      // create filtered list of valid intellisense options
+      if (this.latestIdent) {
+        this.intellisenseOpts = Array.from(
+          this.intellisenseState.boundIdentifiers()
+        ).filter((g) =>
+          g.variableName.startsWith(
+            this.latestIdent?.ident.replace(/[{} \\]/g, "") ?? ""
+          )
+        );
+
+        // sort the intellisense options so that closer ones appear first
+        const listModel = Calc.controller.listModel;
+        const orderMap = new Map<string, number>();
+        for (let i = 0; i < listModel.drawOrder.length; i++) {
+          orderMap.set(listModel.drawOrder[i], i);
+        }
+        const selectedId = getSelectedExpressionID();
+        if (selectedId) {
+          const myindex = orderMap.get(selectedId) ?? 0;
+          this.intellisenseOpts.sort((a, b) => {
+            return (
+              Math.abs((orderMap.get(a.exprId) ?? 0) - myindex) -
+              Math.abs((orderMap.get(b.exprId) ?? 0) - myindex)
+            );
+          });
+        }
+
+        if (this.intellisenseIndex === -1) this.intellisenseIndex = 0;
+      } else {
+        this.intellisenseOpts = [];
+        this.intellisenseIndex = -1;
+      }
+
+      // if there isn't, just get rid of the intellisense window
+    } else {
+      this.intellisenseOpts = [];
+      this.latestIdent = undefined;
+      this.latestMQ = undefined;
+      this.intellisenseIndex = -1;
+    }
+
+    // update intellisense window
+    this.view?.update();
+  }
+
+  // leave an intellisense menu and return to whatever expression
+  // you were previously in
+  leaveIntellisenseMenu() {
+    if (this.intellisenseReturnMQ) {
+      this.intellisenseReturnMQ?.focus();
+      this.latestMQ = this.intellisenseReturnMQ;
+    }
+
+    if (this.prevCursorPos && this.latestMQ) {
+      const mqRootBlock =
+        this.latestMQ.__controller.container.querySelector(
+          ".dcg-mq-root-block"
+        );
+
+      if (!mqRootBlock) return;
+
+      const mqRootBlockRect = mqRootBlock.getBoundingClientRect();
+
+      // simulate a click to get cursor in the right spot
+      const eventHandlerSettings = {
+        bubbles: true,
+        clientX:
+          this.prevCursorPos.x - mqRootBlock.scrollLeft + mqRootBlockRect.x,
+        clientY:
+          this.prevCursorPos.y - mqRootBlock.scrollTop + mqRootBlockRect.y,
+      };
+      this.latestMQ.__controller.container.dispatchEvent(
+        new MouseEvent("mousedown", eventHandlerSettings)
+      );
+      this.latestMQ.__controller.container.dispatchEvent(
+        new MouseEvent("mouseup", eventHandlerSettings)
+      );
```

Comment:
Does a single click event not work?

Reply:
I just tried this. It doesn't seem to work with only a `"click"` event.